### ISSUE TITLE
"Unable to read Dart source"?

### ISSUE THREAD

**Issue by Hixie** [flutter/flutter#6523]
I don't think I was doing anything particularly noteworthy, I just tried to start a random app I'd written, and got the following (nothing ran on the device as far as I saw):

```
$ flutter --local-engine 'android_debug_unopt' run --pid-file /tmp/run.pid
Running 'pub get' in dummy...                        483ms
Building APK in debug mode (android_debug_unopt)...      
Warning: signing the APK using the debug keystore.
Built build/app.apk (9.2MB).
Launching loader on Nexus 6...
--------- beginning of system
W/ActivityManager: Permission Denial: Accessing service ComponentInfo{com.google.android.music/com.google.android.music.dial.DialMediaRouteProviderService} from pid=9788, uid=10090 that is not exported from uid 10070
Observatory listening on http://127.0.0.1:8100
Diagnostic server listening on http://127.0.0.1:8101
W/ActivityManager: Scheduling restart of crashed service com.google.android.gms/.feedback.FeedbackService in 1000ms
I/flutter : [INFO:SkFontMgr_android.cpp(163)] Requested font file /system/fonts/DroidSansFallback.ttf does not exist or cannot be opened.
I/flutter : 
Syncing files to device...                           3198ms
Synced 5.2MB.
Connected to _flutterView/0xa80163b0.
Running lib/main.dart on Nexus 6...
E/flutter : [ERROR:../../lib/tonic/file_loader/file_loader.cc(155)] error: Unable to read Dart source 'package:flutter/rendering.dart'.
W/flutter : LEAK: 20 RenderObject
Application finished.
W/ActivityManager: Force removing ActivityRecord{aac47b2 u0 com.yourcompany.dummy/org.domokit.sky.shell.SkyActivity t363}: app died, no saved state
W/ActivityManager: Unable to start service Intent { act=com.urbanairship.location.ACTION_CHECK_LOCATION_UPDATES cmp=uk.co.economist/com.urbanairship.location.LocationService (has extras) } U=0: not found
```

cc @johnmccutchan

**Comment by johnmccutchan** [comment]
@Hixie was this a one-off failure or repeatable? We may have a race where we are still writing files to DevFS while we are launching.

**Comment by Hixie** [comment]
I only noticed it once yesterday.

**Comment by johnmccutchan** [comment]
I haven't seen this in along time. 

One thing of note is that on Android we are using a directory that the system can choose to clear at anytime.

**Comment by Hixie** [comment]
Going to close this since I don't see it anymore. Please reopen if it's something for which we could add defense in depth and make the error case more graceful.

**Comment by cbracken** [comment]
I suspect @Hixie didn't have any spaces or characters other than plain alpha-numerics in his paths, but for future archaeologists, this typically occurs when:
1. paths contain spaces, accented characters, etc.
2. an imported file is missing when the vm tries to load it

The fix to 1 is tracked by #6577, fixed in the engine by flutter/engine#4650.
